File: app/src/main/java/task/ui/com/testui/adapter/NewsAdapter.java
package task.ui.com.testui.adapter;

import android.content.Context;
import android.media.Image;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.ArrayList;
import java.util.List;

import de.hdodenhof.circleimageview.CircleImageView;
import task.ui.com.testui.R;
import task.ui.com.testui.model.News;

public class NewsAdapter extends RecyclerView.Adapter<NewsAdapter.RowHolder> {
    ArrayList<News> newsList;
    Context context;

    public NewsAdapter(ArrayList<News> newsList, Context context) {
        this.newsList = newsList;
        this.context = context;
    }

    @Override
    public NewsAdapter.RowHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        LayoutInflater inflater = LayoutInflater.from(parent.getContext());
        View holder = inflater.inflate(R.layout.holder_news_item,parent,false);
        return new RowHolder(holder);
    }

    @Override
    public void onBindViewHolder(NewsAdapter.RowHolder holder, int position) {
        String newsTitle = newsList.get(position).getTitle();
        String newsLikes = "Fajri and "+newsList.get(position).getLikers()+" others";
        int comments = newsList.get(position).getComment();
        String newsAuthorName = newsList.get(position).getAuthor_name();
//        int resAuthor = context.getResources().getIdentifier(newsList.get(position).getAuthor_photo() , "drawable", context.getPackageName());
//        int resNews = context.getResources().getIdentifier(newsList.get(position).getPhoto() , "drawable", context.getPackageName());
//        Picasso.get().load(resAuthor).into(holder.civNewsAuthorPhoto);
//        Picasso.get().load(resNews).into(holder.ivNewsPhoto);
        holder.tvNewsTitle.setText(newsTitle);
        holder.tvNewsAuthorName.setText(newsAuthorName);
        holder.tvNewsLikers.setText(newsLikes);
        holder.tvNewsComments.setText(String.valueOf(comments));
    }

    @Override
    public int getItemCount() {
        return newsList.size();
    }

    public class RowHolder extends RecyclerView.ViewHolder {
        TextView tvNewsAuthorName, tvNewsTitle, tvNewsLikers, tvNewsComments;
        ImageView ivNewsPhoto;
        CircleImageView civNewsAuthorPhoto;
        public RowHolder(View itemView) {
            super(itemView);
            tvNewsAuthorName = itemView.findViewById(R.id.tv_news_author_name);
            tvNewsTitle = itemView.findViewById(R.id.tv_news_title);
            tvNewsComments = itemView.findViewById(R.id.tv_news_totalcomment);
            tvNewsLikers = itemView.findViewById(R.id.tv_news_likes);
            ivNewsPhoto = itemView.findViewById(R.id.iv_news_photo);
            civNewsAuthorPhoto = itemView.findViewById(R.id.civ_news_author_photo);
        }
    }
}

File: app/src/main/java/task/ui/com/testui/adapter/NewsAdapter.kt
package task.ui.com.testui.adapter

import android.content.Context
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.TextView

import java.util.ArrayList

import de.hdodenhof.circleimageview.CircleImageView
import task.ui.com.testui.R
import task.ui.com.testui.model.News


class NewsAdapter(val newsList: ArrayList<News>, val context: Context) : RecyclerView.Adapter<NewsAdapter.RowHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup?, viewType: Int): RowHolder {
        val v = LayoutInflater.from(parent?.context).inflate(R.layout.holder_news_item,parent,false)
        return RowHolder(v)
    }

    override fun getItemCount(): Int {
        return newsList.size
    }

    override fun onBindViewHolder(holder: RowHolder?, position: Int) {
        val newsTitle = newsList.get(position).title
        val newsLikes = "Fajri and " + newsList[position].likers + " others"
        val comments = newsList[position].comment
        val newsAuthorName = newsList[position].author_name
        val idPhoto: Int = context.resources.getIdentifier(newsList[position].author_photo, "drawable",
                context.getPackageName())
        val newsPhoto: Int = context.resources.getIdentifier(newsList[position].photo, "drawable", context.packageName)
        holder?.tvNewsTitle?.text = newsTitle
        holder?.tvNewsAuthorName?.text = newsAuthorName
        holder?.tvNewsLikers?.text = newsLikes
        holder?.tvNewsComments?.text = comments.toString()
        holder?.civNewsAuthorPhoto?.setImageResource(idPhoto)
        holder?.ivNewsPhoto?.setImageResource(newsPhoto)
    }

    class RowHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        val tvNewsAuthorName = itemView.findViewById<TextView>(R.id.tv_news_author_name)
        val tvNewsTitle = itemView.findViewById<TextView>(R.id.tv_news_title)
        val tvNewsComments = itemView.findViewById<TextView>(R.id.tv_news_totalcomment)
        val tvNewsLikers = itemView.findViewById<TextView>(R.id.tv_news_likes)
        val ivNewsPhoto = itemView.findViewById<ImageView>(R.id.iv_news_photo)
        val civNewsAuthorPhoto = itemView.findViewById<CircleImageView>(R.id.civ_news_author_photo)
    }

}